sdk/azidentity/azure_cli_credential.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+// Enables authentication to Azure Active Directory using Azure CLI to generated an access token.
+type AzureCLICredential struct {
+	Client      *azureCLICredentialClient
+	shellClient shellClient
+}
+
+// AzureCLICredentialOptions contains the parameters for the ShellClient to execute the Azure CLI command that will return an Access Token.
+type AzureCLICredentialOption struct {
+	shellClientOption shellClient
+}
+
+// NewAzureCLICredential creates an instance of AzureCLICredential to authenticate against Azure Active Directory with Azure CLI Credential's token.
+func NewAzureCLICredential(options *AzureCLICredentialOption) *AzureCLICredential {
+	var Client = newAzureCLICredentialClient()
```

Comment:
local vars should start with lowercase. Plus, this code is very confusing to me. It creates a *Client which is than passed to create an *Options and then you grab just 1 field (shellClientOption) out of the options struct? AND, on top of this, the client that is pass to set the default options is then ignored in the returned options but used to initialize the AzireCLICredential's Client field?

In addition, the way that default options are being initialized/obtained is not in line with our guidelines.